integrations/access/servicenow/app.go

```diff
@@ -226,27 +236,52 @@ func (a *App) onWatcherEvent(ctx context.Context, event types.Event) error {
 }
 
 func (a *App) onPendingRequest(ctx context.Context, req types.AccessRequest) error {
-	if len(req.GetSystemAnnotations()) == 0 {
-		logger.Get(ctx).Debug("Cannot proceed further. Request is missing any annotations")
-		return nil
+	reqID := req.GetName()
+	log := logger.Get(ctx).WithField("reqId", reqID)
+
+	resourceNames, err := a.BaseApp.GetResourceNames(ctx, req)
+	if err != nil {
+		return trace.Wrap(err)
 	}
 
-	// First, try to create a notification incident.
-	isNew, notifyErr := a.tryNotifyService(ctx, req)
+	reqData := RequestData{
+		User:              req.GetUser(),
+		Roles:             req.GetRoles(),
+		RequestReason:     req.GetRequestReason(),
+		SystemAnnotations: req.GetSystemAnnotations(),
+		Resources:         resourceNames,
+	}
+
+	// Create plugin data if it didn't exist before.
+	isNew, err := a.modifyPluginData(ctx, reqID, func(existing *PluginData) (PluginData, bool) {
+		if existing != nil {
+			return PluginData{}, false
+		}
+		return PluginData{RequestData: reqData}, true
+	})
+	if err != nil {
+		return trace.Wrap(err, "updating plugin data")
+	}
 
 	// To minimize the count of auto-approval tries, let's only attempt it only when we have just created an incident.
 	// But if there's an error, we can't really know if the incident is new or not so lets just try.
-	if !isNew && notifyErr == nil {
+	if !isNew {
 		return nil
 	}
-	// Don't show the error if the annotation is just missing.
-	if errors.Is(notifyErr, errMissingAnnotation) {
-		notifyErr = nil
+
+	log.Infof("Creating servicenow incident for: %v", reqID)
+	if err = a.createIncident(ctx, reqID, reqData); err != nil {
+		return trace.Wrap(err, "creating ServiceNow incident")
+	}
+
+	if reqReviews := req.GetReviews(); len(reqReviews) > 0 {
+		if err = a.postReviewNotes(ctx, reqID, reqReviews); err != nil {
+			return trace.Wrap(err, "posting review notes")
+		}
```

Comment:
Just to ensure I'm following the logic here
If we have at least one review, we don't try to auto-approve. Is that it?

Reply:
That section is just to post the review notes if there are some already by the time the plugin gets to that access request.

Reply:
You are right, I'm sorry